src/distrs/beta.jl

```diff
@@ -12,5 +11,7 @@ import .RFunctions:
     betainvlogccdf
 
 # pdf
-betapdf(α::Float64, β::Float64, x::Float64) = exp(betalogpdf(α, β, x))
-betapdf(α::Real, β::Real, x::Real) = betapdf(Float64(α), Float64(β), Float64(x))
+betapdf(α::Real, β::Real, x::Number) = x^(α - 1) * (1 - x)^(β - 1) / beta(α, β)
+
+# logpdf
+betalogpdf(α::Real, β::Real, x::Number) = (α - 1) * log(x) + (β - 1) * log(1 - x) - lbeta(α, β)
```

Comment:
`log1p` is for precision. The number density around zero is much higher than one and `log1p` maps a small neighborhood of zero to a small neighborhood of zero.
```
julia> log1p(eps()/2)
1.1102230246251565e-16
```
vs
```
julia> log(1 + eps()/2) # (= log(1.0))
0.0
```
Before we can consider this PR, we'd have to check the precision compared to the implementations in Rmath.